multicluster/controllers/multicluster/stale_controller.go

```diff
@@ -416,19 +407,36 @@ func (c *StaleResCleanupController) Run(stopCh <-chan struct{}) {
 	defer klog.InfoS("Shutting down StaleResCleanupController")
 
 	ctx, _ := wait.ContextForChannel(stopCh)
-	if err := c.RunOnce(ctx); err != nil {
-		c.Enqueue()
-		go wait.UntilWithContext(ctx, c.runWorker, time.Second)
+
+	switch c.clusterRole {
+	case LeaderCluster:
+		go wait.UntilWithContext(ctx, c.cleanupMemberClusterAnnounces, memberClusterAnnounceStaleTime/2)
+	case MemberCluster:
+		if err := c.RunOnce(ctx); err != nil {
+			c.Enqueue()
+			go func() {
```

Comment:
Does it need to this complex? I think `wait.UntilWithContext` allows you to stop the routinue via the context?

Reply:
Yeah, refine the codes with a cancel function.